src/AutoRoller.sol

```diff
@@ -291,12 +295,15 @@ contract AutoRoller is ERC4626 {
             yt.collect(); // Burns the YTs.
         }
 
+        targetedRate = utils.getNewTargetedRate(targetedRate, address(adapter), maturity, space);
```

Comment:
Why is the Targeted rate calculated in `startCooldown` and not `roll`? If it could be either, I recommend it be in `roll` so that the series rollers pay for the computation.

Reply:
"We delete all data related to the previous series in the startCooldown. This decision is more for gas optimization reasons, but in the future, we can reconsider this design assumption" - @jparklev